### ISSUE TITLE
Not receiving messages after a while 

### ISSUE THREAD

**Issue by matiasdecarli** [mqttjs/MQTT.js#851]
Hello, I'm experiencing an issue with this library

I have a node.js app running on a Raspberry Pi. The connection is done successfully and I can send / receive messages, but after a while (sometimes 20 minutes, sometimes 4 hours), I stop receiving messages (even when I'm connected to that machine via `ssh`)

I'm running this application in several machines and not all of them experience the same issue.

"aws-iot-device-sdk": "2.2.0"
"mqtt" : "2.15.1"
node --version: 8.10.0

Sometimes the app goes offline 20 / 30 minutes and after he comes back, this is the error that I see

```
{ Error: read EHOSTUNREACH
    at _errnoException (util.js:1022:11)
    at TLSWrap.onread (net.js:628:25)
  code: 'EHOSTUNREACH',
  errno: 'EHOSTUNREACH',
  syscall: 'read',
  type: 'error',
  target:
   WebSocket {
     domain: null,
     _events: {},
     _eventsCount: 0,
     _maxListeners: undefined,
     readyState: 3,
     bytesReceived: 2206,
     extensions: null,
     protocol: 'mqttv3.1',
     _binaryType: 'arraybuffer',
     _finalize: { [Function: bound finalize] __ultron: 0 },
     _closeFrameReceived: false,
     _closeFrameSent: false,
     _closeMessage: '',
     _closeTimer: null,
     _finalized: true,
     _closeCode: 1006,
     _receiver: null,
     _sender: null,
     _socket: null,
     _ultron: null,
     protocolVersion: 13,
     _isServer: false,
     url: 'wss://[edited].amazonaws.com/mqtt?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=[edited]&X-Amz-Date=[edited]&X-Amz-SignedHeaders=host&X-Amz-Signature=[edited]',
     _req: null } }
Error: read EHOSTUNREACH
    at _errnoException (util.js:1022:11)
    at TLSWrap.onread (net.js:628:25)
[debug][closeDeviceConn] going down...
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! Exit status 1
npm ERR!
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-05-17T23_10_50_944Z-debug.log
```

I'm connecting using this options (my use case don't require offline queuing)
```
protocolVersion: 4,
connectTimeout: 5 * 1000,
offlineQueueing: false,
offlineQueueMaxSize: 1,
drainTimeMs: 50,
protocol: 'wss',
keepalive: 5,
host: AWS_IOT_ENDPOINT,
clientId: DEVICE_NAME
```

Any help to find the root cause & solution would be highly appreciated

**Comment by BrunnerLivio** [comment]
I also do not receive any messages after a while. Do not know if it is a MQTT.js problem though. Running the client on an Azure VM inside of a Docker container.

**Comment by matiasdecarli** [comment]
For further reference, and now that i've solved my issue, I'm going to comment what was happening to me

I was running the client in a node app inside Docker, using the Docker netowork. This caused the app to have a different IP from the host IP. I started to look at using static IP's for the host but none of those things worked. I even tried using [this](https://docs.docker.com/compose/compose-file/#extra_hosts) and manually setting the IP of the iot endopoint 

At the end, I've tried [using the same network as the host](https://docs.docker.com/network/host/), and this did the trick. Now is working like a charm
